siege_utilities/geo/providers/census_geocoder.py

```
@@ -0,0 +1,397 @@
+        return bg[:12] if len(bg) >= 12 else ""
+
+
+def _get_geocoder(vintage: CensusVintage = CensusVintage.CURRENT):
+    """Get a censusgeocode.CensusGeocode instance with the specified vintage."""
+    try:
+        import censusgeocode
+    except ImportError:
+        raise ImportError(
+            "censusgeocode is required for Census geocoding. "
+            "Install it with: pip install 'siege-utilities[geo]' or pip install censusgeocode"
+        )
```

Ruff (0.15.11)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)